x-pack/plugins/security_solution/server/lib/detection_engine/signals/build_events_query.ts

```diff
@@ -99,22 +125,43 @@ export const buildEventsSearchQuery = ({
       ],
       ...(aggregations ? { aggregations } : {}),
       sort: [
-        {
-          [sortField]: {
-            order: sortOrder ?? 'asc',
-            unmapped_type: 'date',
-          },
-        },
+        ...(timestampOverride != null
+          ? [
+              {
+                [timestampOverride]: {
+                  order: sortOrder ?? 'asc',
+                  unmapped_type: 'date',
+                },
+              },
+              {
+                '@timestamp': {
+                  order: sortOrder ?? 'asc',
+                  unmapped_type: 'date',
+                },
+              },
+            ]
+          : [
+              {
+                '@timestamp': {
+                  order: sortOrder ?? 'asc',
+                  unmapped_type: 'date',
+                },
+              },
+            ]),
       ],
     },
   };
 
-  if (searchAfterSortId) {
+  if (
+    searchAfterSortIds != null &&
+    !isEmpty(searchAfterSortIds) &&
+    searchAfterSortIds.filter((sortId) => !isEmpty(sortId?.toString())).length > 0
```

Comment:
when would we get back empty or undefined values for some of the sort IDs? seems like we would want all the sort IDs or else we'd have a problem here, since the array of search after values has to correspond to the array provided to the sort parameter of the query

Reply:
So oddly enough the type for sort id's is [defined to allow](https://github.com/elastic/elasticsearch-js/blob/6464fc69f13f259ed813a922af42089de45354c8/api/types.d.ts#L12005) `null` so I think this is leftover logic from me playing around with that. I will update this though.

I tried a few different ways to represent this "null" value in dev tools to see what I would get back and couldn't determine when the sort fields would return `null`. I'm thinking I'll change this code to remove the above filter and when the sortId is `null` or `> Number.MAX_SAFE_INTEGER` then just send over that stringified Java max_int.

My understanding from the issues I've read around missing sort fields is Elasticsearch will default to returning that Java Max_int value whenever it can't find something for the sort fields. Good catch.

Reply:
updated here: d29c16f277ddcd2de34316709e669e30498feb53

Reply:
sounds like a good approach, thanks for the testing and update :+1: